server/src/main/java/org/apache/cassandra/sidecar/tasks/CdcRawDirectorySpaceCleaner.java

```diff
@@ -0,0 +1,467 @@
+        }
+
+        LOGGER.warn(
+        "Could not read cdc_total_space from system_views.settings, falling back to props");
+        return cdcConfiguration.fallbackCdcRawDirectoryMaxSizeBytes();
+    }
+
+    @Override
+    public ScheduleDecision scheduleDecision()
+    {
+        if (cdcConfiguration.enableCdcRawDirectoryRoutineCleanUp())
+        {
+            return ScheduleDecision.EXECUTE;
+        }
+        LOGGER.debug("Skipping CdcRawDirectorySpaceCleaner: feature is disabled");
+        return ScheduleDecision.SKIP;
+    }
+
+    protected void routineCleanUp()
+    {
+        List<String> dataDirectories = instanceMetadata.dataDirs();
+        dataDirectories.stream()
+                       .map(dir -> new File(dir, CDC_DIR_NAME))
+                       .forEach(this::cleanUpCdcRawDirectory);
+    }
+
+    protected void cleanUpCdcRawDirectory(File cdcRawDirectory)
+    {
+        if (!cdcRawDirectory.exists() || !cdcRawDirectory.isDirectory())
+        {
+            LOGGER.debug("Skipping CdcRawDirectorySpaceCleaner: CDC directory does not exist: " +
+                         cdcRawDirectory);
+            return;
+        }
+
+        List<CdcRawSegmentFile> segmentFiles = Optional
+                                               .ofNullable(
+                                               cdcRawDirectory.listFiles(this::validSegmentFilter))
+                                               .map(files -> Arrays.stream(files)
+                                                                   .map(CdcRawSegmentFile::new)
+                                                                   .filter(
+                                                                   CdcRawSegmentFile::indexExists)
+                                                                   .collect(Collectors.toList())
+                                               )
+                                               .orElseGet(List::of);
+        publishCdcStats(segmentFiles);
+        if (segmentFiles.size() < 2)
+        {
+            LOGGER.debug(
+            "Skipping cdc data cleaner routine cleanup: No cdc data or only one single cdc segment is found.");
+            return;
+        }
+
+        long directorySize = FileUtils.directorySize(cdcRawDirectory);
+        long upperLimitBytes =
+        (long) (maxUsage() * cdcConfiguration.cdcRawDirectoryMaxPercentUsage());
+        // Sort the files by segmentId to delete commit log segments in write order
+        // The latest file is the current active segment, but it could be created before the retention duration, e.g. slow data ingress
+        Collections.sort(segmentFiles);
+        long nowInMillis = timeProvider.currentTimeMillis();
+
+        // track the age of the oldest commit log segment to give indication of the time-window buffer available
+        cdcMetrics.oldestSegmentAge.metric.setValue(
+        (int) MILLISECONDS.toMinutes(nowInMillis - segmentFiles.get(0).lastModified()));
```

Comment:
`oldestSegmentAge` has the name `CdcRawOldestSegmentAgeSeconds`. The unit is seconds, not minutes.